staging/src/k8s.io/cli-runtime/pkg/genericclioptions/resource/builder_test.go

```diff
@@ -569,7 +569,98 @@ func TestDirectoryBuilder(t *testing.T) {
 	}
 }
 
+func setupKustomizeDirectory() (string, error) {
+	path, err := ioutil.TempDir("/tmp", "")
+	if err != nil {
+		return "", err
+	}
+
+	contents := map[string]string{
+		"configmap.yaml": `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: the-map
+data:
+  altGreeting: "Good Morning!"
+  enableRisky: "false"
+`,
+		"deployment.yaml": `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: the-deployment
+spec:
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        deployment: hello
+    spec:
+      containers:
+      - name: the-container
+        image: monopole/hello:1
+        command: ["/hello",
+                  "--port=8080",
+                  "--enableRiskyFeature=$(ENABLE_RISKY)"]
+        ports:
+        - containerPort: 8080
+        env:
+        - name: ALT_GREETING
+          valueFrom:
+            configMapKeyRef:
+              name: the-map
+              key: altGreeting
+        - name: ENABLE_RISKY
+          valueFrom:
+            configMapKeyRef:
+              name: the-map
+              key: enableRisky
+`,
+		"service.yaml": `
+kind: Service
+apiVersion: v1
+metadata:
+  name: the-service
+spec:
+  selector:
+    deployment: hello
+  type: LoadBalancer
+  ports:
+  - protocol: TCP
+    port: 8666
+    targetPort: 8080
+`,
+		"kustomization.yaml": `
+nameprefix: test-
+resources:
+- deployment.yaml
+- service.yaml
+- configmap.yaml
+`,
+	}
+
+	for filename, content := range contents {
+		f, err := os.Create(filepath.Join(path, filename))
```

Comment:
nit, but you could use [`ioutil.WriteFile`](https://golang.org/pkg/io/ioutil/#WriteFile)

Reply:
updated